File: abc/LockerDoors/src/Lambda.rb
def bruteForce(n)
    arr = Array.new(n, false)

    1.upto(n) do |i|
        (i - 1).step(n - 1, i) do |j|
            arr[j] = !arr[j]
        end
    end

    arr
end

def process(n)
    arr = Array.new(n)

    temp = 0
    1.upto(n) do |i|
        temp = Math.sqrt(i.to_f()).to_i()
        arr[i - 1] = temp * temp == i
    end

    arr
end

if __FILE__ == $PROGRAM_NAME
    N = 100

    arr = process(N)

    arr.each_index do |i|
        if arr[i]
            puts(format("%d\t  OPEN", i + 1))
        end
    end
end

File: abc/LockerDoors/src/Lambda.java
public class Lambda {

    public static boolean[] bruteForce(int n) {
        boolean[] arr = new boolean[n];

        for (int i = 1; i <= n; i++) {
            for (int j = i - 1; j < n; j += i) {
                arr[j] = !arr[j];
            }
        }

        return arr;
    }

    public static boolean[] process(int n) {
        boolean[] arr = new boolean[n];

        int temp;
        for (int i = 1; i <= n; i++) {
            temp = (int) Math.sqrt((double) i);
            arr[i - 1] = temp * temp == i;
        }

        return arr;
    }

    public static void main(String[] args) {
        final int N = 100;

        boolean[] arr = process(N);

        for (int i = 0; i < N; i++) {
            if (arr[i]) {
                System.out.printf("%d\t  OPEN%n", i + 1);
            }
        }
    }
}
